migration-script/replace_css_examples.py

```diff
@@ -0,0 +1,165 @@
+import json
+import os
+import html
+from bs4 import BeautifulSoup
+
+# Directory to search for all index.md files
+input_files_pattern = "./files/**/index.md"
+meta_files_pattern = "../interactive-examples/live-examples/css-examples/**/meta.json"
+
+interactive_examples_folder = "../interactive-examples"
+
+meta_files = glob.glob(meta_files_pattern, recursive=True)
+meta_map = {}
+
+for meta_file in meta_files:
+    with open(meta_file, "r") as file:
+        data = json.load(file)
+        for value in data["pages"].values():
+            meta_map[value["fileName"]] = value
+
+# Replace the macro with actual code blocks
+
+
+def replace_macros(content, md_file):
+    def replace_macro(match):
+        built_path = match.group(1)
+        if path_match := re.search(r"^pages\/css\/(.*)$", built_path):
+            filename = path_match.group(1)
+            try:
+                meta = meta_map[filename]
+                example_code_path = os.path.join(interactive_examples_folder,
+                                                 meta["exampleCode"])
+            except KeyError:
+                print(f"No such file: {filename}")
+                return match.group(0)  # keep the macro unchanged
+
+            try:
+                css_example_src_path = os.path.join(interactive_examples_folder,
+                                                    meta["cssExampleSrc"])
+            except KeyError:
+                css_example_src_path = None
+            try:
+                js_example_src_path = os.path.join(interactive_examples_folder,
+                                                   meta["jsExampleSrc"])
+            except KeyError:
+                js_example_src_path = None
+
+            with open(example_code_path, "r") as file:
+                example_code = file.read()
+                soup = BeautifulSoup(example_code, 'html.parser')
+                css_choices = [code.get_text(strip=True) for code in soup.find_all('code', class_='language-css')]
+                html_example_src = soup.find('section', id='default-example').decode_contents().strip()
+
+            if css_example_src_path:
+                with open(css_example_src_path, "r") as file:
+                    css_example_src = file.read()
+                    if "url(" in css_example_src:
+                        css_example_src = map_media(css_example_src)
+            else:
+                css_example_src = None
+
+            if js_example_src_path:
+                with open(js_example_src_path, "r") as file:
+                    js_example_src = file.read()
+                    if "media/" in js_example_src or "interactive-examples.mdn.mozilla.net" in js_example_src:
+                        js_example_src = map_media(js_example_src)
+            else:
+                js_example_src = None
+
+
+            css_choice_fences = "\n\n".join(
+                f"""```css interactive-example-choice
+{css_choice.replace("&amp;shy;", "&shy;").rstrip()}
```

Comment:
Shouldn't need the `.replace` here, was just for a single weirdly written html example